mava/utils/jumanji_registry.py

```diff
@@ -0,0 +1,134 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from typing import Any, Callable, Dict
+
+# Define the ranges of valid attributes
+LBF_ATTRIBUTES_RANGES: Dict[str, Any] = {
+    "grid_size": range(5, 20),  # Grid size, the height and width are equal.
+    "fov": range(2, 20),  # Field of view of agent.
+    "num_agents": range(2, 20),
+    "num_food": range(1, 10),
+    "force_coop": [True, False],  # Force cooperation between agents.
+}
+
+RWARE_ATTRIBUTES_RANGES: Dict[str, Any] = {
+    "num_agents": range(1, 20),
+    "difficulties": {"-easy": 2, "": 1, "-hard": 0.5},
+    "size": {
+        "tiny": (1, 3),
+        "small": (2, 3),
+        "medium": (2, 5),
+        "large": (3, 5),
+    },
+}
+
+
+def get_task_config(env_name: str, scenario: str) -> Any:
+    """
+    Retrieves the attributes for a given environment and scenario.
+
+    Args:
+        env_name (str): The name of the environment.
+        scenario (str): The scenario to retrieve attributes for.
+
+    Returns:
+        dict: A dictionary containing the attributes of the scenario.
+    """
+    if env_name not in ENV_FUNCTIONS:
+        raise ValueError(f"Unknown environment name: {env_name}")
+
+    # Extract and return scenario attributes
+    env_fn = ENV_FUNCTIONS[env_name]
+    return env_fn(scenario)
+
+
+def get_lbf_config(scenario: str) -> Dict[str, Any]:
+    """
+    Checks if a scenario is valid and extracts its attributes for LevelBasedForaging.
+
+    Args:
+        scenario (str): The scenario to retrieve attributes for.
+
+    Returns:
+        dict: A dictionary containing the attributes of the scenario.
+    """
+    # Extract attributes based on valid ranges. E.g. "2s-10x10-3p-3f-coop"
+    grid_size = int(scenario.split("x")[0].split("-")[-1])
+    fov = int(scenario.split("s")[0]) if "s" in scenario else grid_size
+    num_agents = int(scenario.split("p")[0].split("-")[-1])
+    num_food = int(scenario.split("f")[0][-1])
+    force_coop = "coop" in scenario.split("-")
+
+    scenario_attributes = {
+        "grid_size": grid_size,
+        "fov": fov,
+        "num_agents": num_agents,
+        "num_food": num_food,
+        "force_coop": force_coop,
+    }
+
+    # Validate attributes against the predefined ranges
+    valid_attributes = all(
+        LBF_ATTRIBUTES_RANGES[key].__contains__(value) for key, value in scenario_attributes.items()
+    )
+
+    if not valid_attributes:
+        raise ValueError(f"Invalid attributes for scenario: {scenario}")
+
+    # Get and return the scenario attributes
+    scenario_attributes["max_agent_level"] = 2
```

Comment:
Is this always true for LBF, can you not have an agent level above 2?

Reply:
On the Jumanji side we have the flexibility to change that but in the [original implementation](https://github.com/semitable/lb-foraging/blob/60939b921e8e9f8ab5affa33c4ad29e916b47d41/lbforaging/__init__.py#L16) it's set to 2.
 In this case we can't extract this from the scenario name, we have to add it manually in the code 😞.

Reply:
That's fine, was just checking if that's what the original implementation does :+1: